controller/array_action/array_mediator_svc.py

```diff
@@ -720,12 +723,23 @@ def create_snapshot(self, volume_id, snapshot_name, space_efficiency, pool):
         logger.info("finished creating snapshot '{0}' from volume '{1}'".format(snapshot_name, volume_id))
         return self._generate_snapshot_response(target_cli_volume, source_cli_volume.vdisk_UID)
 
-    def delete_snapshot(self, snapshot_id):
+    def _is_addsnapshot_supported(self):
+        return hasattr(self.client.svctask, "addsnapshot")
+
+    def _rmsnapshot(self, internal_snapshot_id):
+        try:
+            self.client.svctask.rmsnapshot(snapshotid=internal_snapshot_id)
+        except (svc_errors.CommandExecutionError, CLIFailureError) as ex:
+            if SNAPSHOT_NOT_EXIST in ex.my_message:
+                raise array_errors.ObjectNotFoundError(internal_snapshot_id)
+            raise ex
```

Comment:
should not fail [if exception is just a warning](https://github.com/IBM/ibm-block-csi-driver/pull/538#discussion_r917422095)